### PR TITLE
boost a search result based on Addon.is_recommended

### LINKED ISSUE #11066: Update search to prioritize recommended add-ons
Search should boost recommended add-ons above non-recommended content.

See the PRD for further details: https://docs.google.com/document/d/1fRxLy7h02tn1HBIlsTU_6h9DHmN8c3N6bg4JoxHF7J8/edit#heading=h.5kvm14zbadi2

### LINKED ISSUE #11410: Don't raise an exception if there isn't a `current_version` in `is_recommended()`
### Describe the problem and steps to reproduce it:
An add-on with no `current_version` (either purely unlisted or listed with all its versions deleted) that we're trying to determine the recommendation status of can cause us to raise an exception.

### Steps to reproduce
- Create a purely unlisted add-on, make it recommendable by adding a discovery item for it through the admin w/ the property checked.
- Also try with a purely listed add-on for which all versions are deleted after it has been approved & set as recommendable in discovery admin
- Try to index it, or try to call the detail API for that add-on w/ the right permissions 

### What did you expect to happen?
- It should work. Notably, once indexed the add-on should no longer show up in searches if it was listed before.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #11066 by adding a x20 (actually x5, compared to non-recommended, non-experimental add-ons) boost to recommended add-ons.  Not sure if that's enough, but we can increase it.
+fixes #11410

### CODE REVIEW

**Comment by diox** [comment]
Looks like I kept the field for backwards-compatibility but forgot to file an issue to then remove it later. My bad... Gotta check that we did a full reindex since that date before removing it. In any case the recommended boost should be done at query time by tweaking the scoring functions.

**Comment by EnTeQuAk** [comment]
> Wait, I actually did change that to a query time boost a while ago as part of #10089 (fixing #10071). Did I forget to remove the indexer part ?

I knew I remembered something but couldn't find it so I thought I rememberd wrong. Thanks for clarifying!

File: src/olympia/search/tests/test_filters.py [hunk 1]
```diff
@@ -121,6 +121,10 @@ def _test_q(self, qs):
             },
             'weight': 4.0
         }
+        assert functions[2] == {
+            'filter': {
+                'term': {'is_recommended': True}},
```
Comment (diox) [inline src/olympia/search/tests/test_filters.py]: Kind of an edge case but I think we need to repeat the other boosts in `functions[1]` as well to ensure we don't boost an add-on that is somehow `is_recommended` but not public ? Not sure if that's possible.

Comment (eviljeff) [reply src/olympia/search/tests/test_filters.py]: It's possible for a non-public add-on to be `is_recommended` but a non-public add-on wouldn't be in the search results so doesn't need bumping (that's my thinking).

**Pull request merged**